src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/ThemeManager3.cs

```diff
@@ -0,0 +1,452 @@
+                {
+                    ApplyFluentOnWindow(window);
+                }
+            }
+        }
+    }
+
+    internal static void OnApplicationThemeChanged(ThemeMode oldThemeMode, ThemeMode newThemeMode)
+    {
+        IgnoreAppResourcesChange = true;
+
+        try
+        {
+            if(newThemeMode == ThemeMode.None)
+            {
+                if(oldThemeMode != newThemeMode)
+                {
+                    RemoveFluentFromApplication();
+                }
+                return;
+            }
+
+            bool useLightColors = GetUseLightColors(newThemeMode);
+            var fluentThemeResourceUri = GetFluentThemeResourceUri(useLightColors);
+            AddOrUpdateThemeResources(Application.Current.Resources, fluentThemeResourceUri);
+
+            foreach(Window window in Application.Current.Windows)
+            {
+                // Replace this with a check for the window theme
+                if(!FluentEnabledWindows.HasItem(window))
+                {
+                    ApplyStyleOnWindow(window, useLightColors);
+                }
+            }
+        }
+        finally
+        {
+            IgnoreAppResourcesChange = false;
+        }
+    }
+
+    internal static void OnWindowThemeChanged(Window window, ThemeMode oldThemeMode, ThemeMode newThemeMode)
+    {
+        if(newThemeMode == ThemeMode.None)
+        {
+            if(newThemeMode != oldThemeMode)
+            {
+                RemoveFluentFromWindow(window);
+            }
+            return;
+        }
+
+        ApplyFluentOnWindow(window);
+    }
+
+    internal static bool SyncThemeModeAndResources()
+    {
+        if(DeferSyncingThemeModeAndResources) return true;
+
+        ResourceDictionaryContainsFluentDictionary(Application.Current.Resources, out ThemeMode themeMode);
+
+        if(Application.Current.ThemeMode != themeMode)
+        {
+            Application.Current.ThemeMode = themeMode;
+            return true;
+        }
+        return false;
+    }
+
+    internal static void SyncDeferredThemeModeAndResources()
+    {
+        if(Application.Current == null) return;
+
+        ThemeMode themeMode = Application.Current.ThemeMode;
+
+        bool resyncThemeMode = false;
+        int index = FindLastFluentThemeResourceDictionaryIndex(Application.Current.Resources);
+
+        if(index > 0)
+        {
+            ResourceDictionaryContainsFluentDictionary(Application.Current.Resources, out ThemeMode _themeMode);            
+            resyncThemeMode = true;
+            themeMode = _themeMode;
+        }
+        else
+        {
+            if(themeMode != ThemeMode.None && index != 0)
+            {
+                resyncThemeMode = true;
+            }
+
+            if(themeMode == ThemeMode.None && index == 0)
+            {
+                resyncThemeMode = true;
+                ResourceDictionaryContainsFluentDictionary(Application.Current.Resources, out ThemeMode _themeMode);            
```

Comment:
Assign directly into `themeMode`?
```suggestion
                ResourceDictionaryContainsFluentDictionary(Application.Current.Resources, out themeMode);         
```